pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -157,8 +161,14 @@ while [ "$#" -gt 0 ]; do
     esac
 done
 
-if [[ -n "$SUDO_USER" || -n $remoteSudo ]]; then
-    maybeSudo=(sudo --preserve-env="$preservedSudoVars" --)
+sudoCommand=(sudo --preserve-env="$preservedSudoVars" --)
+
+if [[ -n "$SUDO_USER" ]]; then
+    maybeSudo=("${sudoCommand[@]}")
+fi
+
+if [[ -n "$remoteSudoAlways" ]]; then
+    maybeSudo=(-t "${sudoComand[@]}")
```

Comment:
This seems to rely on a coincidence that `maybeSudo` gets pasted onto some other command consistently.
Please don't do that.

Reply:
I've changed it to always use `-t` with `ssh` as a PTY may also be necessary with non-`sudo` commands e.g. connecting for the first time and SSH asking if you want to trust an unknown host